tensorboard/plugins/debugger/tensor_helper.py

```diff
@@ -82,25 +82,42 @@ def array_view(array, slicing=None, mapping=None):
   """
 
   dtype = str(array.dtype)
+  # String-type TensorFlow Tensors are represented as object-type arrays in
+  # numpy. We map the type name back to 'string' for clarity.
+  if dtype == 'object':
+    dtype = 'string'
   sliced_array = (array[command_parser._parse_slices(slicing)] if slicing
                   else array)
-  shape = sliced_array.shape
-  if mapping == "image/png":
-    if len(sliced_array.shape) == 2:
-      return dtype, shape, array_to_base64_png(sliced_array)
-    elif len(sliced_array.shape) == 3:
-      raise NotImplementedError(
-          "image/png mapping for 3D array has not been implemented")
-    else:
-      raise ValueError("Invalid rank for image/png mapping: %d" %
-                       len(sliced_array.shape))
-  elif mapping == 'health-pill':
-    health_pill = health_pill_calc.calc_health_pill(array)
-    return dtype, shape, health_pill
-  elif mapping is None or mapping == '' or  mapping.lower() == 'none':
-    return dtype, shape, sliced_array.tolist()
+
+  if np.isscalar(sliced_array) and str(dtype) == 'string':
+    # When a string Tensor (for which dtype is 'object') is sliced down to only
```

Comment:
Does this logic exist because `numpy.reshape(sliced_array, array.shape)` could separate the string into several strings?

Reply:
The logic in this special case is because in the case of a string scalar, `sliced` array is just a Python string, i.e., not a numpy array. This is specific to the scalar case, i.e., if you have vector of strings with length > 1, you still get a numpy array.